assignment1/Exercise1.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python
+
+from string import ascii_lowercase
+
+def is_permutation(string1, string2) :
+    """Determines if two strings are permutations of each other.
+    
+    Args:
+        string1: the first string to compare
+        string2: the second string to compare
+        
+    Returns:
+        True if the two strings are permutations of each other, otherwise false
+    """
+    if string1 is None or string2 is None:
+        return False
+    
+    string1_dict = string_to_dict(string1)
+    string2_dict = string_to_dict(string2)
+    
+    return string1_dict == string2_dict
+
+def is_permutation_english(string1, string2) : 
+    if string1 is None or string2 is None:
+        return False
+    
+    string1_dict = string_to_dict_english(string1)
+    string2_dict = string_to_dict_english(string2)
+    
+    if string1_dict == {} or string2_dict == {} :
+        return False
+    
+    return string1_dict == string2_dict
+    
+def string_to_dict(input_string) :
+    lower_case_string = input_string.lower()
+    
+    # create a dictionary where you map each character with a counter of
+    # how many instances of a character exist
+    dictionary = {}
+    
+    for character in lower_case_string:
+        if character in dictionary:
+            dictionary[character] += 1
```

Comment:
You can use the `get` function to remove the if statement.
For example:
```python
  dictionary.get('A', 42)
```
will return the value stored in the dictionary for the string `'A'` if present, or `42` if not present.
Of course, `'A'` and `42` are just examples.